packages/block-library/src/navigation/edit/menu-inspector-controls.js

```diff
@@ -34,15 +35,97 @@ const BLOCKS_WITH_LINK_UI_SUPPORT = [
 ];
 const { PrivateListView } = unlock( blockEditorPrivateApis );
 
+/**
+ * Shared cleanup function for auto-inserted Navigation Link blocks.
+ *
+ * Removes the block if it has no URL and clears the inserted block state.
+ * This ensures consistent cleanup behavior across different contexts.
+ *
+ * @param {Object}   insertedBlock    - The currently inserted block data
+ * @param {Function} removeBlock      - Function to remove a block
+ * @param {Function} setInsertedBlock - Function to clear the inserted block state
+ */
+function cleanupInsertedBlock( insertedBlock, removeBlock, setInsertedBlock ) {
+	// Prevent automatic block selection when removing blocks in list view context
+	// This avoids focus stealing that would close the list view and switch to canvas
+	const shouldAutoSelectBlock = false;
+
+	// Follows the exact same pattern as Navigation Link block's onClose handler
+	// If there is no URL then remove the auto-inserted block to avoid empty blocks
+	if ( ! insertedBlock?.attributes?.url && insertedBlock?.clientId ) {
+		// Remove the block entirely to avoid poor UX
+		// This matches the Navigation Link block's behavior
+		removeBlock( insertedBlock.clientId, shouldAutoSelectBlock );
+	}
+	setInsertedBlock( null );
+}
+
+/**
+ * Custom hook to handle iframe interaction cleanup for LinkUI in List View context.
+ *
+ * PROBLEM: When LinkUI is open in the List View and the user clicks on a block in the
+ * editor canvas (which is in an iframe), the Popover's built-in focus detection doesn't
+ * work because iframe events don't bubble up to the parent document. This causes the
+ * auto-inserted Navigation Link block to remain in the canvas instead of being cleaned up.
+ *
+ * SOLUTION: When LinkUI is open, we add a mousedown listener to the editor canvas iframe.
+ * When triggered, we call the provided cleanup function to handle block removal.
+ *
+ * @param {boolean}  isLinkUIOpen   - Whether the LinkUI is currently rendered
+ * @param {Object}   insertedBlock  - The currently inserted block data
+ * @param {Function} cleanupFn      - Function to call for cleanup when iframe is clicked
+ */
+function useEditorCanvasCleanup( isLinkUIOpen, insertedBlock, cleanupFn ) {
```

Comment:
@ciampo I had to add this manual detect for clicks within the editor canvas because useDialog's focus detection wasn't firing `onClose` on `Popover` for clicks within the `iframe`. I wonder if you've seen a problem like this before. If not should we be looking to fix in `useDialog`?

Reply:
Looks like a similar problem to https://github.com/WordPress/gutenberg/issues/58825